internal/renew_root.go

```diff
@@ -75,3 +93,102 @@ func (s fetchRootStep) Execute(h *rootRenewalContext) error {
 	}
 	return nil
 }
+
+func loadCertsFromPem(data []byte) (certs []*x509.Certificate, err error) {
+	var block *pem.Block
+	if len(data) == 0 {
+		return nil, errors.New("Unexpected empty PEM block")
+	}
+	for len(data) > 0 {
+		if block, data = pem.Decode(data); block == nil {
+			return nil, fmt.Errorf("Malformed PEM block at %s", data[:100])
+		} else if block.Type != "CERTIFICATE" {
+			return nil, fmt.Errorf("Invalid PEM block type: %s", block.Type)
+		} else if c, err := x509.ParseCertificate(block.Bytes); err != nil {
+			return nil, fmt.Errorf("Invalid X.509 certificate: %w", err)
+		} else {
+			certs = append(certs, c)
+		}
+	}
+	return
+}
+
+// As per crypto.PublicKey documentation, all public keys implement this interface.
+type publicKey interface {
+	Equal(x crypto.PublicKey) bool
+}
+
+func validateRootCerts(curCerts, newCerts []*x509.Certificate) error {
+	// Each new certificate must pass all of the below checks:
+	// 1. It is a valid certificate authority.
```

Comment:
do we need to do any checking of certificate `NotBefore` or `NotAfter`?

Reply:
TBH I wanted this validation to be simple.

There are valid use cases when a user might want to e.g. set `NotBefore` for a new CA into the future, to make sure that this new CA cannot be spoiled somehow before they tested everything and make a decision to do the phase 2 of the CA renewal.
They might even decide to test the phase 1 of the rotation with the `NotAfter` in the past, until they are certain that their rotation workflow is secure enough.

Reply:
That said, I think of our API side, the validation is going to be more strict, as we are going to provide this _secure enough_ end-2-end workflow.
But, there are 2 things to factor in:
- There is no need to copy all the complex validation rules to the device side.
- Plus, we need to support "custom" EST server implementations, which might have different business needs.